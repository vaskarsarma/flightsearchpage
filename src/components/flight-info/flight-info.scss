.Flight-info {
  display: flex;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
  margin: 10px;
}

@media screen and (min-width: 1100px) {
  .Flight-info img {
    margin-right: 4%;
  }
  .Flight-info .detail-label:nth-child(2),
  .Flight-info .detail-label:nth-child(3),
  .Flight-info .detail-label:nth-child(4),
  .Flight-info .detail-label:nth-child(5),
  .Flight-info .detail-label:nth-child(6) {
    flex: 1 1 9%;
    text-align: left;
  }
  .Flight-info Button {
    margin-left: 4%;
    flex: 1 1 10%;
  }
}

@media screen and (min-width: 601px) and (max-width: 1099px) {
  .Flight-info .detail-label:nth-child(2) {
    flex: 5 1 80%;
    text-align: end;
    margin-bottom: 15px;
  }
  .Flight-info .detail-label:nth-child(3),
  .Flight-info .detail-label:nth-child(4) {
    flex: 2 1 20%;
  }
  .Flight-info .detail-label:nth-child(5) {
    flex: 2 1 20%;
  }
  .Flight-info .detail-label:nth-child(6) {
    flex: 2 1 20%;
  }
  .Flight-info Button {
    flex: 1 1 70%;
  }
}

@media screen and (min-width: 401px) and (max-width: 600px) {
  .Flight-info .detail-label:nth-child(2) {
    flex: 5 1 80%;
    text-align: end;
    margin-bottom: 15px;
  }
  .Flight-info .detail-label:nth-child(3),
  .Flight-info .detail-label:nth-child(4) {
    flex: 2 1 33%;
  }
  .Flight-info .detail-label:nth-child(5) {
    flex: 2 1 11%;
    text-align: end;
  }
  .Flight-info .detail-label:nth-child(6),
  .Flight-info .detail-label:nth-child(7) {
    flex: 2 1 33%;
  }
  .Flight-info Button {
    flex: 2 1 15%;
  }
}

@media screen and (min-width: 231px) and (max-width: 400px) {
  .Flight-info .detail-label:nth-child(2) {
    flex: 5 1 80%;
    text-align: end;
    margin-bottom: 15px;
  }
  .Flight-info .detail-label:nth-child(3) {
    flex: 1 1 50%;
    text-align: left;
  }
  .Flight-info .detail-label:nth-child(4) {
    flex: 1 1 50%;
    text-align: end;
  }
  .Flight-info .detail-label:nth-child(5) {
    flex: 1 1 50%;
    text-align: left;
  }
  .Flight-info .detail-label:nth-child(6) {
    flex: 1 1 50%;
    text-align: end;
  }
  .Flight-info .detail-label:nth-child(7) {
    flex: 1 1 50%;
    text-align: left;
  }
  .Flight-info Button {
    flex: 1 1 50%;
  }
}

@media screen and (max-width: 230px) {
  .Flight-info .detail-label:nth-child(2),
  .Flight-info .detail-label:nth-child(3),
  .Flight-info
    .detail-label:nth-child(4)
    .Flight-info
    .detail-label:nth-child(5),
  .Flight-info .detail-label:nth-child(6),
  .Flight-info .detail-label:nth-child(7) {
    flex: 1 1 100%;
    text-align: left;
    margin-bottom: 10px;
  }
  .Flight-info Button {
    flex: 1 1 80%;
  }
}
