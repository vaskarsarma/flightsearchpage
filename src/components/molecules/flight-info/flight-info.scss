.Flight-info {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-between;
	align-items: center;
	padding: 10px;
	margin: 10px;
}
@media screen and(min-width: 1100px) {
	.Flight-info {
		img {
			margin-right: 4%;
		}
		.detail-label {
			&:nth-child(2) {
				flex: 1 1 9%;
				text-align: left;
			}
			&:nth-child(3) {
				flex: 1 1 9%;
				text-align: left;
			}
			&:nth-child(4) {
				flex: 1 1 9%;
				text-align: left;
			}
			&:nth-child(5) {
				flex: 1 1 9%;
				text-align: left;
			}
			&:nth-child(6) {
				flex: 1 1 9%;
				text-align: left;
			}
		}
		Button {
			margin-left: 4%;
			flex: 1 1 10%;
		}
	}
}
@media screen and(min-width: 601px) and(max-width: 1099px) {
	.Flight-info {
		.detail-label {
			&:nth-child(2) {
				flex: 5 1 80%;
				text-align: end;
				margin-bottom: 15px;
			}
			&:nth-child(3) {
				flex: 2 1 20%;
			}
			&:nth-child(4) {
				flex: 2 1 20%;
			}
			&:nth-child(5) {
				flex: 2 1 20%;
			}
			&:nth-child(6) {
				flex: 2 1 20%;
			}
		}
		Button {
			flex: 1 1 70%;
		}
	}
}
@media screen and(min-width: 401px) and(max-width: 600px) {
	.Flight-info {
		.detail-label {
			&:nth-child(2) {
				flex: 5 1 80%;
				text-align: end;
				margin-bottom: 15px;
			}
			&:nth-child(3) {
				flex: 2 1 33%;
			}
			&:nth-child(4) {
				flex: 2 1 33%;
			}
			&:nth-child(5) {
				flex: 2 1 11%;
				text-align: end;
			}
			&:nth-child(6) {
				flex: 2 1 33%;
			}
			&:nth-child(7) {
				flex: 2 1 33%;
			}
		}
		Button {
			flex: 2 1 15%;
		}
	}
}
@media screen and(min-width: 231px) and(max-width: 400px) {
	.Flight-info {
		.detail-label {
			&:nth-child(2) {
				flex: 5 1 80%;
				text-align: end;
				margin-bottom: 15px;
			}
			&:nth-child(3) {
				flex: 1 1 50%;
				text-align: left;
			}
			&:nth-child(4) {
				flex: 1 1 50%;
				text-align: end;
			}
			&:nth-child(5) {
				flex: 1 1 50%;
				text-align: left;
			}
			&:nth-child(6) {
				flex: 1 1 50%;
				text-align: end;
			}
			&:nth-child(7) {
				flex: 1 1 50%;
				text-align: left;
			}
		}
		Button {
			flex: 1 1 50%;
		}
	}
}
@media screen and(max-width: 230px) {
	.Flight-info {
		.detail-label {
			&:nth-child(2) {
				flex: 1 1 100%;
				text-align: left;
				margin-bottom: 10px;
			}
			&:nth-child(3) {
				flex: 1 1 100%;
				text-align: left;
				margin-bottom: 10px;
			}
			&:nth-child(6) {
				flex: 1 1 100%;
				text-align: left;
				margin-bottom: 10px;
			}
			&:nth-child(7) {
				flex: 1 1 100%;
				text-align: left;
				margin-bottom: 10px;
			}
			&:nth-child(4) .Flight-info .detail-label:nth-child(5) {
				flex: 1 1 100%;
				text-align: left;
				margin-bottom: 10px;
			}
		}
		Button {
			flex: 1 1 80%;
		}
	}
}
