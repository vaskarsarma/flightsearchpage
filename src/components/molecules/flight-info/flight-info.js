import React from "react";
import { Card, Button } from "react-bootstrap";
import { DetailLabel } from "../../atoms/detail-label/detail-label";
import FlightLogo from "../../atoms/flightlogo/flightlogo";
import nonStopFlightLogo from "../../../assets/nonstop.png";
import "./flight-info.scss";

export const FlightInfo = (props) => {
  const { id, flightInfoObj, disableBtn, onClickedhandler } = props;
  const depratBy = flightInfoObj.date + " " + flightInfoObj.departureTime;
  const arrivedBy = flightInfoObj.date + " " + flightInfoObj.arrivalTime;
  const price = "$" + flightInfoObj.price;

  return (
    <div className="Cart-Container">
      <Card
        bg={disableBtn ? "warning" : ""}
        border={disableBtn ? "danger" : ""}
      >
        <section className="Flight-info">
          <FlightLogo
            imagePath={nonStopFlightLogo}
            alttext="Icon"
            size="Flight-Icon"
          ></FlightLogo>
          <DetailLabel
            mainText="AIRLINE"
            subText={flightInfoObj.name}
          ></DetailLabel>
          <DetailLabel
            mainText="FROM"
            subText={flightInfoObj.origin}
          ></DetailLabel>
          <DetailLabel
            mainText="TO"
            subText={flightInfoObj.destination}
          ></DetailLabel>
          <DetailLabel mainText="DEPART BY" subText={depratBy}></DetailLabel>
          <DetailLabel mainText="ARRIVED AT" subText={arrivedBy}></DetailLabel>
          <DetailLabel mainText="PRICE" subText={price}></DetailLabel>
          <Button
            id={id}
            variant="primary"
            onClick={onClickedhandler}
            disabled={disableBtn}
          >
            SELECT
          </Button>
        </section>
      </Card>
    </div>
  );
};
