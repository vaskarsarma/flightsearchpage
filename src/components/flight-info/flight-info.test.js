import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { FlightInfo } from "./flight-info";
import mockRoutes from "../mocks/mockroutes.json";

describe("<FlightInfo />", () => {
  let container;
  const activeRoute = mockRoutes[0];
  const nonActiveRoute = mockRoutes[1];

  beforeEach(() => {
    container = shallow(<FlightInfo {...activeRoute} />);
  });

  it("<FlightInfo /> Rendered Correctly", () => {
    expect(shallowToJson(container)).toMatchSnapshot();
  });

  it("should render a div", () => {
    expect(container.find("div").length).toEqual(1);
    expect(container.find("div.Cart-Container").length).toBe(1);
  });

  it("should render a Card", () => {
    expect(container.find("Card").length).toEqual(1);
  });

  it("should highlight the Card if button is disabled", () => {
    expect(container.find("Card").props()["bg"]).toEqual("warning");
    expect(container.find("Card").props()["border"]).toEqual("danger");
  });

  it("should not highlight the Card if button is active", () => {
    container.setProps({ ...nonActiveRoute });
    expect(container.find("Card").props()["bg"]).toEqual("");
    expect(container.find("Card").props()["border"]).toEqual("");
  });
});
