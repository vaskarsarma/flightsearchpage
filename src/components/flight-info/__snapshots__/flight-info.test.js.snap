// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlightInfo /> <FlightInfo /> Rendered Correctly 1`] = `
<div
  className="Cart-Container"
>
  <Card
    bg="warning"
    body={false}
    border="danger"
  >
    <section
      className="Flight-info"
    >
      <FlightLogo
        alttext="Image Icon"
        imagePath="nonstop.png"
        size="Flight-Icon"
      />
      <DetailLabel
        mainText="AIRLINE"
        subText="Air India"
      />
      <DetailLabel
        mainText="FROM"
        subText="Mumbai (BOM)"
      />
      <DetailLabel
        mainText="TO"
        subText="Delhi (DEL)"
      />
      <DetailLabel
        mainText="DEPART BY"
        subText="2020/11/02 14:30"
      />
      <DetailLabel
        mainText="ARRIVED AT"
        subText="2020/11/02 16:30"
      />
      <DetailLabel
        mainText="PRICE"
        subText="$4170"
      />
      <Button
        active={false}
        disabled={true}
        id={1}
        onClick={1}
        variant="primary"
      >
        SELECT
      </Button>
    </section>
  </Card>
</div>
`;
